File: go/Structures/LinkedList/linkedlist.go
package main

import "fmt"

// Create Node struct

type Node struct {
	data int
	next *Node
}

type LinkedList struct {
	head *Node
}

func (list *LinkedList) Insert(value int) {
	newNode := &Node{data: value}
	if list.head == nil {
		list.head = newNode
	} else {
		current := list.head
		for current.next != nil {
			current = current.next
		}
		current.next = newNode
	}
}

func (list *LinkedList) Search(value int) {
	current := list.head
	for current != nil {
		if current.data == value {
			fmt.Println("Value found: ", current.data, *current)
			return
		} else {
			current = current.next
		}
	}
	fmt.Println("Value Not found")
}

func (list *LinkedList) Display() {
	current := list.head

	for current != nil {
		fmt.Printf("%v --> ", current.data)
		current = current.next
	}
	fmt.Println("")
}

func (list *LinkedList) Delete(value int) {
	current := list.head
	for current != nil {
		next := current.next.next
		if current.next.data == value {
			current.next = next
			return
		}
		current = current.next
	}
}

func main() {
	x := LinkedList{}

	for i := 0; i < 100; i++ {
		x.Insert(i)
	}
	x.Display()
}

File: c/Structures/LinkedList/linkedlist.c
#include <stdio.h>
#include <stdlib.h>
typedef struct Node {
  int data;
  struct Node *next;
} Node;

typedef struct LinkedList {
  Node *head;
} LinkedList;

void InsertList(LinkedList *list, int data) {
  Node *newNode = (Node *)malloc(sizeof(Node));
  if (newNode == NULL) {
    return;
  }
  newNode->data = data;
  newNode->next = NULL;

  if (list->head == NULL) {
    list->head = newNode;
    return;
  } else {
    Node *current = list->head;
    while (current->next != NULL) {

      current = current->next;
    }
    current->next = newNode;
    return;
  }
}

void DestroyList(LinkedList *list) {
  Node *current = list->head;
  Node *next;
  while (current != NULL) {
    next = current->next;
    free(current);

    current = next;
  }
  free(list);
}
void PrintList(LinkedList *list) {
  Node *current = list->head;
  while (current != NULL) {
    printf("%d->", current->data);
    current = current->next;
  }
  printf("NULL\n");
}
int main(void) {
  LinkedList *list = malloc(sizeof(LinkedList));
  for (int i = 1; i <= 100; i++) {
    InsertList(list, i);
  }
  PrintList(list);

  DestroyList(list);
  return 1;
}
